src/cloud_provider/aws/databases/redis.rs

```diff
@@ -577,3 +569,25 @@ impl Backup for Redis {
         unimplemented!()
     }
 }
+
+fn get_redis_version(
+    requested_version: &str,
+    is_managed_service: bool,
+) -> Result<String, StringError> {
+    let mut supported_redis_versions = HashMap::with_capacity(2);
+
```

Comment:
I think you should check the input. The version must be 5 or 6 in this case. Other versions should throw an error.

Reply:
It's done by the function called get_supported_version_to_use in this function. If it doesn't match, I throws an error saying:
this {} version is not supported